### PR TITLE
feat(experience): Added dropdown layout plugin

### PR DESCRIPTION
**Pull request by tobi-or-not-tobi**
The dropdown layout plugin is used to render components inside a dropdown navigation. 

closes: [HRZ-90385](https://spryker.atlassian.net/browse/HRZ-90385)

[HRZ-90385]: https://spryker.atlassian.net/browse/HRZ-90385?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: libs/base/ui/overlays/dropdown/dropdown.styles.ts [hunk 1]
```diff
@@ -23,7 +23,7 @@ export const dropdownBaseStyles = css`
       var(--oryx-popover-maxwidth)
     );
 
-    z-index: 100;
+    /* z-index: 100; */
```
Comment (dunqan) [inline libs/base/ui/overlays/dropdown/dropdown.styles.ts]: what's the resason for this one?

File: libs/platform/experience/src/services/layout/plugins/types/dropdown/dropdown.plugin.ts [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+export class DropdownLayoutPlugin implements LayoutPlugin {
+  getConfig(): Observable<LayoutPluginConfig> {
+    return of({
+      schema: () => import('./dropdown.schema').then((m) => m.schema),
+    });
+  }
+
+  getStyles(data: LayoutPluginOptionsParams): Observable<LayoutStyles> {
+    // TODO: add isComposition to data set so we can better check
+    // console.log(data, data.options.layout !== 'dropdown');
```
Comment (dunqan) [inline libs/platform/experience/src/services/layout/plugins/types/dropdown/dropdown.plugin.ts]: leftover?

File: libs/platform/experience/src/services/layout/plugins/types/navigation/navigation-layout.styles.ts [hunk 3]
```diff
@@ -18,6 +18,10 @@ export const styles: LayoutStyles = {
       flex-wrap: wrap;
       align-self: stretch;
     }
+
+    oryx-composition {
```
Comment (dunqan) [inline libs/platform/experience/src/services/layout/plugins/types/navigation/navigation-layout.styles.ts]: just wondering if it should be decided from the outside 🤔

File: libs/base/ui/overlays/dropdown/dropdown.styles.ts (hunk 1 above)
Comment (tobi-or-not-tobi) [reply libs/base/ui/overlays/dropdown/dropdown.styles.ts]: This has been a mistake from the beginning. I've however wrapped it inside a featureVersion to keep BC.

**Review by dunqan — APPROVED** [review]

**Pull request merged**